w3af/plugins/output/json_file.py

```diff
@@ -0,0 +1,145 @@
+"""
+json_file.py
+
+Copyright 2012 Andres Riancho
+
+This file is part of w3af, http://w3af.org/ .
+
+w3af is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+w3af is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+"""
+import os
+import base64
+import json
+
+import w3af.core.data.kb.knowledge_base as kb
+import w3af.core.controllers.output_manager as om
+
+from w3af.core.controllers.plugins.output_plugin import OutputPlugin
+from w3af.core.data.options.opt_factory import opt_factory
+from w3af.core.data.options.option_types import OUTPUT_FILE
+from w3af.core.data.options.option_list import OptionList
+
+class json_file(OutputPlugin):
+    """
+    Export identified vulnerabilities to a JSON file.
+
+    :author: jose nazario (jose@monkey.org)
+    """
+
+    def __init__(self):
+        OutputPlugin.__init__(self)
+        self.output_file = '~/output-w3af.json'
+    
+    def do_nothing(self, *args, **kwargs):
+        pass
+
+    debug = log_http = vulnerability = do_nothing
+    information = error = console = log_enabled_plugins = do_nothing
+
+    def end(self):
+        self.flush()
+
+    def flush(self):
+        """
+        Exports the vulnerabilities and information to the user configured
+        file.
+        """
+        self.output_file = os.path.expanduser(self.output_file)
+
+        try:
+            output_handler = file(self.output_file, 'wb')
+        except IOError, ioe:
+            msg = 'Failed to open the output file for writing: "%s"'
+            om.out.error(msg % ioe)
+            return
+
+        items = []
```

Comment:
I think that at least we should have two items in the json document root:
 * scan info
 * items

The scan info should be similar to the XML, containing the target URL and (also important) the JSON document version. This last bit of information is helpful for parsers to identify how to handle the json document

Items should have all the vulns